### PR TITLE
[Components with more than 1 service] LPS-200589 Refactor ElasticsearchConnectionManager component and ElasticsearchClientResolver interface

### PR DESCRIPTION
**Pull request by rcsdnj**
LPS-200589 portal-search-elasticsearch7-api - Unify ElasticsearchClientResolver and ElasticsearchConnectionManager under the same interface, avoiding ElasticSearchManager being registered in more than 1 service

Context: ElasticSearchManager was implementing the ElasticsearchClientResolver interface. There was no other implementation of ElasticsearchClientResolver (except for test fixtures), so all the points which were using the ElasticsearchClientResolver reference were changed to use ElasticsearchConnectionManager (which was refactored to have its own interface and implementation)

Revised by Rafael Praxedes in https://github.com/rafaprax/liferay-portal/pull/1337
Incorporated @dantewang suggestions from https://github.com/liferay-core-infra/liferay-portal/pull/6180#issuecomment-1825411949

### CODE REVIEW

**Comment by catarineaguiar** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
ci:test:search

**Comment by zhangyan0701** [comment]
No regressions found. But the SF failure should be related.
`     [java] The commit message is missing a reference to a required JIRA project: BLADE, CLDSVCS, COMMERCE, FACES, IDE, ISSD, ISSITES, LCD, LHC, LOOP, LPS, LRAC, LRCI, LRDOCS, LRIS, LRQA, OAUTH2, POSHI, RELEASE, SYNC, TR. Please verify that the JIRA project keys are specifiedin source-formatter.properties in the liferay-portal repository.`

There seems a blank space at the beginning of the commit message of https://github.com/liferay-core-infra/liferay-portal/pull/6228/commits/824ead77fa92ab40b75f5d85e0e3be5af08ff23c.

**Comment by rcsdnj** [comment]
ci:test:sf

**Comment by rcsdnj** [comment]
ci:test:sf

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ElasticsearchClientResolver.java [hunk 1]
```diff
@@ -19,4 +39,12 @@ public interface ElasticsearchClientResolver {
 	public RestHighLevelClient getRestHighLevelClient(
 		String connectionId, boolean preferLocalCluster);
 
+	public boolean isCrossClusterReplicationEnabled();
+
+	public void removeElasticsearchConnection(String connectionId);
+
+	@Activate
+	public void activate();
```
[the remark this answers is not in the document]
Comment (rcsdnj) [reply modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ElasticsearchClientResolver.java]: Thanks, Prax. I've fixed it

**Comment by rcsdnj** [comment]
ci:test:sf

**Comment by rcsdnj** [comment]
ci:test:relevant

**Review by rafaprax — APPROVED** [review]

**Review by rafaprax — APPROVED** [review]

**Comment by rafaprax** [comment]
ci:test:sf

**Comment by rafaprax** [comment]
Hey @dantewang, 

In the last force push, @rcsdnj just moved some changes, SF ones, for the first commit. I mean, there are no changes in the code, just the way they were committed, see https://github.com/liferay-core-infra/liferay-portal/compare/21d3041b22b0361f2c3bb5ddb455af09f0faf6fd..de0bd55649620767637c9de958dc576d5eda8c82. So, the last QA analysis still valid.

Thx

**Comment by dantewang** [comment]
@rafaprax It conflicts with Shuyang's recent changes, please re-apply and resend, thanks!

**Pull request closed without merging**